src/legacy/core_plugins/kibana/public/discover/kibana_services.ts

```diff
@@ -50,13 +50,15 @@ export function setServices(newServices: any) {
 // EXPORT legacy static dependencies, should be migrated when available in a new version;
 export { angular };
 export { wrapInI18nContext } from 'ui/i18n';
-export { getRequestInspectorStats, getResponseInspectorStats } from '../../../data/public';
+import { search } from '../../../../../plugins/data/public';
+export const { getRequestInspectorStats, getResponseInspectorStats, tabifyAggResponse } = search;
+// @ts-ignore
+export { shortenDottedString } from '../../common/utils/shorten_dotted_string';
```

Comment:
Is `shorten_dotted_string` becoming an implementation detail of the `data` plugin? This PR is switching  the discover implementation to the legacy kibana plugin implementation which will probably go away. Should this be inlined in discover as well? Or will there be another plugin of some kind exposing the functionality in general?

Reply:
Yeah, since it is a one-line function my thought was to make it an implementation detail of the data plugin, and let Discover inline the function if needed.

AFAICT Discover is the only place outside of Data that uses this function, so it didn't feel worth adding something so small to the public contract, unless you think its usage will become more widespread.

Reply:
Makes sense, we will inline when moving discover over 👍